src/net/sourceforge/kolmafia/textui/parsetree/RecordType.java

```diff
@@ -90,6 +90,12 @@ public Type getIndexType()
 		return DataTypes.STRING_TYPE;
 	}
 
+	@Override
+	public Type getDataType()
```

Comment:
The definition in the abstract class could use a comment describing why it's okay that it's null for a RecordType -- namely that it stores the value type, and records do not necessarily have a single value type for their fields.

(Truthfully, the same can probably be said for all the other abstract methods... but this one in particular stood out to me.)

Reply:
How's that? https://github.com/kolmafia/kolmafia/pull/11/commits/4c510d0f9e4a0f46130ae564951ff2d19d040998